plugins/org.eclipse.n4js.xpect/src/org/eclipse/n4js/xpect/methods/scoping/N4JSCommaSeparatedValuesExpectation.java

```diff
@@ -0,0 +1,249 @@
+			act.setOrdered(annotation.ordered());
+			act.setQuoted(annotation.quoted());
+			act.setSeparator(',');
+			act.setWhitespaceSensitive(annotation.whitespaceSensitive());
+
+			if (actual != null && predicate != null) {
+				if (exp.isWildcard())
+					act.init(exp.applyPredicate(predicate), annotation.itemFormatter());
+				else
+					act.init(actual, annotation.itemFormatter());
+			} else if (predicate != null)
+				act.init(exp.applyPredicate(predicate), annotation.itemFormatter());
+			else if (actual != null)
+				act.init(actual, annotation.itemFormatter());
+			else
+				throw new NullPointerException();
+
+			adaptActual(actual, exp, act);
+
+			String nl = new Text(this.getRegion().getDocument()).getNL();
+			if (!exp.matches(act)) {
+				StringBuilder expString = new StringBuilder();
+				StringBuilder actString = new StringBuilder();
+				boolean expWrap = false;
+				boolean expEmpty = false;
+				boolean actWrap = false;
+				int lineLength = 0, lineCount = 0;
+				for (Pair<Collection<ExpectationItem>, ActualItem> pair : exp.map(act)) {
+					String expItem = null;
+					String actItem = null;
+					if (pair.getFirst() != null && !pair.getFirst().isEmpty()) {
+						if (pair.getSecond() != null)
+							expItem = pair.getSecond().getEscaped();
+						else
+							expItem = pair.getFirst().iterator().next().getEscaped();
+					} else {
+						if (pair.getSecond() != null)
+							expItem = str(pair.getSecond().getEscaped().length());
+					}
+					if (pair.getSecond() != null) {
+						actItem = pair.getSecond().getEscaped();
+						lineCount++;
+						lineLength += actItem.length() + 2;
+						boolean count = annotation.maxItemsPerLine() > 0 && lineCount > annotation.maxItemsPerLine();
+						boolean width = annotation.maxLineWidth() > 0 && lineLength > annotation.maxLineWidth();
+						if (count || width)
+							expWrap = actWrap = true;
+					}
+					if (expItem != null && expString.length() > 0) {
+						if (expWrap) {
+							expString.append(expEmpty ? nl : "," + nl);
+							expWrap = false;
+						} else
+							expString.append(expEmpty ? "  " : ", ");
+					}
+					if (actItem != null && actString.length() > 0) {
+						if (actWrap) {
+							actString.append("," + nl);
+							actWrap = false;
+							lineCount = 0;
+							lineLength = 0;
+						} else
+							actString.append(", ");
+					}
+					if (expItem != null) {
+						expString.append(expItem);
+						expEmpty = expItem.trim().length() == 0;
+					}
+					if (actItem != null)
+						actString.append(actItem);
+				}
+				String expDoc = replaceInDocument(expString.toString());
+				String actDoc = replaceInDocument(actString.toString());
+				throw new ComparisonFailure("", expDoc, actDoc);
+			}
+		}
+
+		/**
+		 * Explicitly request all expected items, as they may appear in scope only on explicit request.
+		 */
+		private void adaptActual(Iterable<?> actual, N4JSExpectationCollection exp, ActualCollection act) {
```

Comment:
I believe this customization is technically not necessary, because it should be possible to wrap the logic from  `adaptActual()` into a `Predicate` and call `expectation.assertEquals(null, predicate)`. It's important that the first parameter is null, otherwise the predicate will only be used on Expectations containing a wildcard.